src/runtime/components/InputNumber.vue

```diff
@@ -161,30 +176,32 @@ defineExpose({
       @focus="emitFormFocus"
     />
 
-    <div :class="ui.increment({ class: props.ui?.increment })">
-      <NumberFieldIncrement as-child :disabled="disabled">
+    <div :class="ui.increment({ class: props.ui?.increment })" @click="onIncrement">
+      <NumberFieldIncrement as-child :disabled="disabled || disabledIncrement">
         <slot name="increment">
           <UButton
             :icon="incrementIcon"
             :color="color"
             :size="size"
             variant="link"
             :aria-label="t('inputNumber.increment')"
+            :disabled="disabled || disabledIncrement"
```

Comment:
No need to specify `disabled` twice, the one on `NumberFieldIncrement` is enough.